### PR TITLE
Tag list

### PR DESCRIPTION
**Pull request by AlexStibbons**
_only_ for showing a tag list; placeholders for clicking on tag

### CODE REVIEW

**Comment by AlexStibbons** [comment]
Should the next branch be joining these two features (note list and tag list) into one?

So, the note list of the new 'join' branch would show a list of notes, where each note in the list also shows a list of tags attached to it. And the tag list in the 'join' branch would enable clicking on a tag in the tag list, after which a list of relevant notes would be shown.

File: app/src/main/java/com/example/goodnote/database/LocalDb.kt [hunk 1]
```diff
@@ -42,21 +40,68 @@ abstract class LocalDb : RoomDatabase() {
     }
 
     private class LocalDbCallback : RoomDatabase.Callback() {
+
         override fun onCreate(db: SupportSQLiteDatabase) {
             super.onCreate(db)
+
             instance?.let { database ->
                 GlobalScope.launch {
```
Comment (vuksa) [inline app/src/main/java/com/example/goodnote/database/LocalDb.kt:50]: We have one issue here. On the first start of the app, the database won't be initialized when you try to fill it with data. 

In order to prefill data on the first app start, you should move this code in the `buildDb` method, and call this logic after room object has been initialized. 

You can also use the second option, where you use `SupportSQLiteDatabase` db parameter of the `onCreate` method and apply raw SQLite commands via this object to insert data to db. 

IMHO,  first option is much easier. :)

File: app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt [hunk 2]
```diff
@@ -27,37 +30,34 @@ class NoteViewModel(private val repository: NoteRepo) : ViewModel() {
...
     }
 
-    fun saveNote(note: Note) = viewModelScope.launch {
+    fun saveNote(note: NoteDetailsModel) = viewModelScope.launch {
 
-        if (note.title.isNullOrEmpty()) note.title = DEFAULT_TITLE
+        val noteSave = if (note.title.isNullOrEmpty()) note.copy(title = DEFAULT_TITLE) else note
 
-        _repoNotes.addOne(note) // --> do not make a new list, but just add a new note (DiffUtil)
+        _repoNotes.addOne(noteSave.toNoteListModel()) // --> do not make a new list, but just add a new note (DiffUtil)
```
Comment (vuksa) [inline app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt]: Implementation of this method is doing exactly what you said you don't want it to do :D
It creates a new list by merging a new object with the old notes list. :)

Please take a closer look at the implementation of this method :)

Comment (AlexStibbons) [reply app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt]: I know, I haven't even touched DiffUtil yet :D

File: app/src/main/java/com/example/goodnote/database/LocalDb.kt (hunk 1 above)
Comment (AlexStibbons) [reply app/src/main/java/com/example/goodnote/database/LocalDb.kt:50]: It weirdly didn't populate with
 `return Room.databaseBuilder(context, LocalDb::class.java, DB_NAME).build().also { populate(it) }`
but did once the `also` wasn't used.

Comment (AlexStibbons) [reply app/src/main/java/com/example/goodnote/database/LocalDb.kt:50]: `private fun buildDb(context: Context): LocalDb {`

`val db = Room.databaseBuilder(context, LocalDb::class.java, DB_NAME).build()`

`populate(db)`

`return db }`

This sometimes populates db, sometimes loads 7/9 notes, sometimes loads tags, sometimes it does not. Reverted back to using `RoomDatabase.Callback()`

File: app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt (hunk 2 above)
Comment (AlexStibbons) [reply app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt]: DiffUtil never changes position/index of items in new list

**Pull request merged**